feature/gribi/ate_tests/hierarchical_weight_resolution_test/hierarchical_weight_resolution_test.go

```diff
@@ -270,51 +269,98 @@ func (a *attributes) configInterfaceDUT(t *testing.T, d *ondatra.DUTDevice, p *o
 		i = a.NewOCInterface(p.Name())
 	}
 
-	a.configSubinterfaceDUT(t, i)
-	intfPath := gnmi.OC().Interface(p.Name())
-	gnmi.Update(t, d, intfPath.Config(), i)
 	if *deviations.ExplicitPortSpeed {
-		fptest.SetPortSpeed(t, p)
+		i.GetOrCreateEthernet().PortSpeed = fptest.GetIfSpeed(t, p)
 	}
+
+	a.configSubinterfaceDUT(t, i)
+	intfPath := gnmi.OC().Interface(p.Name())
+	gnmi.Replace(t, d, intfPath.Config(), i)
 	fptest.LogQuery(t, "DUT", intfPath.Config(), gnmi.GetConfig(t, d, intfPath.Config()))
 }
 
-// configureNetworkInstance creates new Network Instance and configures it, if provided,
-// else configures the Default Network Instance.
-func (a *attributes) configureNetworkInstance(t *testing.T, d *ondatra.DUTDevice, p *ondatra.Port) {
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	// configure NI
+	configureNetworkInstance(t, dut)
+
+	// Configure DUT ports.
+	dutPort1.configInterfaceDUT(t, dut)
+	dutPort2.configInterfaceDUT(t, dut)
+
+	// assign subinterfaces to DEFAULT network instance if needed (deviation-based)
+	dutPort1.assignSubifsToNetworkInstance(t, dut)
+	dutPort2.assignSubifsToNetworkInstance(t, dut)
+
+	// apply PBF to src interface
+	dp1 := dut.Port(t, dutPort1.Name)
+	applyForwardingPolicy(t, dp1.Name())
+}
+
+// configureNetworkInstance creates and configures non-default and default NIs
+func configureNetworkInstance(t *testing.T, d *ondatra.DUTDevice) {
 	t.Helper()
-	// Use default NI if not provided
-	if a.networkInstance != "" {
+
+	// configure non-default VRF
+	for _, vrf := range []string{nonDefaultVRF} {
 		ni := &oc.NetworkInstance{
-			Name: ygot.String(a.networkInstance),
+			Name: ygot.String(vrf),
 			Type: oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_L3VRF,
 		}
-		i := ni.GetOrCreateInterface(p.Name())
-		i.Interface = ygot.String(p.Name())
-		i.Subinterface = ygot.Uint32(0)
-
-		dni := gnmi.OC().NetworkInstance(a.networkInstance)
+		dni := gnmi.OC().NetworkInstance(vrf)
 		gnmi.Replace(t, d, dni.Config(), ni)
 		fptest.LogQuery(t, "NI", dni.Config(), gnmi.GetConfig(t, d, dni.Config()))
-	} else {
-		if *deviations.ExplicitInterfaceInDefaultVRF {
-			dni := gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance)
-			gnmi.Replace(t, d, dni.Interface(p.Name()).Config(), &oc.NetworkInstance_Interface{
-				Id:           ygot.String(p.Name()),
-				Interface:    ygot.String(p.Name()),
-				Subinterface: ygot.Uint32(0),
-			})
+		if *deviations.ExplicitGRIBIUnderNetworkInstance {
+			fptest.EnableGRIBIUnderNetworkInstance(t, d, vrf)
 		}
 	}
+
+	// configure PBF in DEFAULT vrf
+	defNIPath := gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance)
```

Comment:
Can you add the config to specify the instance type for *deviations.DefaultNetworkInstance, like below ?
	gnmi.Replace(t, d, defNIPath.Type().Config(), oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)